--- ../avorion-stock/data/scripts/commands/tmod.lua	1969-12-31 18:00:00.000000000 -0600
+++ ../avorion-turret-editor/data/scripts/commands/tmod.lua	2018-02-17 17:47:21.731899800 -0600
@@ -0,0 +1,15 @@
+
+function execute(PlayerID, Command, Action, ...)
+
+	if(not onServer())
+	then return end
+
+	Player(PlayerID)
+	:removeScript("mods/DccTurretEditor/Commands/TurretModding")
+
+	Player(PlayerID)
+	:addScriptOnce("mods/DccTurretEditor/Commands/TurretModding",Action,...)
+
+	return 0, "", ""
+end
+
